src/modules/job-exec/job-exec.c

```diff
@@ -823,7 +823,10 @@ static flux_future_t *jobinfo_start_init (struct jobinfo *job)
  */
 static int job_get_ns_name (char *buf, int bufsz, flux_jobid_t id)
 {
-    return fluid_encode (buf, bufsz, id, FLUID_STRING_DOTHEX);
+    int n = snprintf (buf, bufsz - 1, "job-%ju", id);
```

Comment:
i think you have to check `n` for validity before doing `n-1` b/c snprintf takes a `size_t`.  although this is internal code, so maybe it doesn't matter.